pkg/services/secrets/manager/manager.go

```diff
@@ -39,15 +41,29 @@ func ProvideSecretsService(
 
 	currentProvider := settings.KeyValue("security", "encryption_provider").MustString(kmsproviders.Default)
 
+	logger := log.New("secrets")
+	if _, ok := providers[currentProvider]; !ok {
```

Comment:
Based on the current implementation of the [osskmsproviders](https://github.com/grafana/grafana/blob/main/pkg/services/kmsproviders/osskmsproviders/osskmsproviders.go#L23), this line will always throw an error with the default configuration because `osskmsproviders.Service` will provide nothing (because the feature toggle is disabled) but this check will look for default provider anyway.

So, we should either move this check behind the feature toggle or make `osskmsproviders.Service` to always provide the default one. In fact, the pipeline failed due to some backend tests failing specifically for this reason (if I'm not wrong).

Reply:
You're right. Thank you!